plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -145,3 +149,20 @@ func getPriorityFunctionConfigs(keys util.StringSet) ([]algorithm.PriorityConfig
 	}
 	return configs, nil
 }
+
+var validName = regexp.MustCompile("^[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])$")
+
+func validateAlgorithmName(name string) error {
```

Comment:
optional nit: is there a reason why this should return an error? isn't a bool sufficient?